doc/whats_new/v1.1.rst

```diff
@@ -121,6 +121,10 @@ Changelog
   :pr:`21448` by :user:`Oleh Kozynets <OlehKSS>` and
   :user:`Christian Ritter <chritter>`.
 
+- |Enhancement| Added warning with feature names for skipped features
+  due to no observed values in :class:`SimpleImputer`.
+  :pr:`21617` by :user:`Christian Ritter <chritter>`.
```

Comment:
Might this phrasing be a little bit more direct?
```suggestion
- |Enhancement| :class:`SimpleImputer` now warns with feature names
  when features which are skipped due to unobserved values.
  :pr:`21617` by :user:`Christian Ritter <chritter>`.
```

Reply:
Or even more explicit:

when features which are skipped due to the lack of any observed values in the training set.

Reply:
Based on @ogrisel's proposal:
```suggestion
- |Enhancement| :class:`SimpleImputer` now warns with feature names
  when features which are skipped due to the lack of any observed
  values in the training set.
  :pr:`21617` by :user:`Christian Ritter <chritter>`.
```